plugins/action/ci_net_map.py

```diff
@@ -0,0 +1,325 @@
+              skip_nm: false
+              mac_addr: a1:69:da:21:aa:03
+              interface_name: eth2
+              ip_v4: 192.168.122.11
+              ip_v6: fdc0:8b54:108a:c949:0000:0000:0000:000b
+              mtu: 1500
+          hostname: instance-2-hostname
+complete_map:
+    description: Tells if the output is the result of a full map (with instances_info)
+    returned: success
+    type: bool
+"""
+
+from ansible.plugins.action import ActionBase
+from ansible.errors import AnsibleActionFail
+
+from ansible_collections.cifmw.general.plugins.module_utils.encoding import (
+    ansible_encoding,
+)
+from ansible_collections.cifmw.general.plugins.module_utils.net_map import (
+    exceptions,
+    networking_mapper,
+)
+
+
+class ActionModule(ActionBase):
+    @staticmethod
+    def __build_interfaces_info_dict(
+        ifaces_info_raw: typing.Dict[str, typing.Any], net_name: str
+    ) -> typing.Dict[str, typing.Any]:
+        ifaces_info_dict = {}
+        for instance_name, ifaces_list in ifaces_info_raw.items():
+            iface_data = next(
+                (
+                    iface_content
+                    for iface_content in ifaces_list
+                    if "network" in iface_content
+                    and iface_content["network"] == net_name
+                ),
+                None,
+            )
+            if instance_name not in ifaces_info_dict and iface_data:
+                ifaces_info_dict[instance_name] = iface_data
+
+        return ifaces_info_dict
+
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        del tmp
+
+        task_args = ansible_encoding.decode_ansible_raw(self._task.args)
+        networking_definition = task_args.get("networking_definition", None)
+        if not networking_definition:
+            raise AnsibleActionFail("networking_definition is a mandatory argument")
+
+        interfaces_info_dict = None
+        is_complete_map = "interfaces_info" in task_args
+        network_name = task_args.get("network_name", None)
+        if is_complete_map and not network_name:
+            raise AnsibleActionFail(
+                "network_name is a mandatory " "argument if interfaces_info is given"
+            )
+        elif is_complete_map:
+            interfaces_info_dict = self.__build_interfaces_info_dict(
+                task_args["interfaces_info"], network_name
+            )
+
+        mapper = networking_mapper.NetworkingDefinitionMapper(
+            dict(task_vars["hostvars"]),
+            task_vars["groups"],
+            options=networking_mapper.NetworkingMapperOptions.from_dict(task_args),
+        )
+
+        try:
+            mapping_result = (
+                mapper.map_complete(networking_definition, interfaces_info_dict)
+                if is_complete_map
+                else mapper.map_partial(networking_definition)
```

Comment:
If `interfaces_info` is not provided `else mapper.map_partial(networking_definition)` handles it? I see `interfaces_info` is optional, what is the use case for a partial map?

Reply:
Yes, it's handled internally properly. That use case is for scenarios where the bootstraping of resources like instances or the networking of the instance requires networking information such as MTU, network ranges, lack or presence of IPv6, etc. For those scenarios the mapper can be called twice, one first call to perform a best effort map without the instances (or part of them only) and a second call with the mac addresses of each instance that will output the entire mapping.